server/routes/wishCards.js

```diff
@@ -199,6 +199,82 @@ router.post(
   },
 );
 
+// @desc    Edit Wishcard
+// @route   PUT '/wishcards/edit/wishcardId'
+// @access  Private, verified partner or admin
+// @tested 	Yes
+router.post(
+  '/edit/:id',
+  limiter,
+  renderPermissions,
+  createWishcardValidationRules(),
+  validate,
+  async (req, res) => {
+    try {
+      let { childBirthday, wishItemPrice } = req.body;
+      const wishcard = await WishCardRepository.getWishCardByObjectId(req.params.id);
+
+      childBirthday = childBirthday ? new Date(childBirthday) : wishcard.childBirthday;
+      wishItemPrice = wishItemPrice ? Number(wishItemPrice) : wishcard.wishItemPrice;
+
+      await WishCardRepository.updateWishCard(wishcard._id, {
+        childBirthday,
+        wishItemPrice,
+        address: {
+          address1: req.body.address1 ? req.body.address1 : wishcard.address.address1,
+          address2: req.body.address2 ? req.body.address2 : wishcard.address.address2,
+          city: req.body.address_city ? req.body.address_city : wishcard.address.address_city,
+          state: req.body.address_state ? req.body.address_state : wishcard.address.address_state,
+          zip: req.body.address_zip ? req.body.address_zip : wishcard.address.address_zip,
+          country: req.body.address_country
+            ? req.body.address_country
+            : wishcard.address.address_country,
+        },
+        ...req.body,
+      });
+
+      let url = '/wishcards/me';
+      if (res.locals.user.userRole === 'admin') {
+        url = '/wishcards/';
+      }
+
+      res.status(200).send({ success: true, url });
+      log.info('Wishcard Updated', { type: 'wishcard_updated', wishCardId: wishcard.id });
+    } catch (error) {
+      handleError(res, 400, error);
+    }
+  },
+);
+
+// @desc    Delete wishcard
+// @route   Delete '/wishcards/delete/wishcardId'
+// @access  Private, verified partner or admin
+// @tested 	Yes
+router.delete('/delete/:id', limiter, renderPermissions, async (req, res) => {
+  try {
+    const wishcard = await WishCardRepository.getWishCardByObjectId(req.params.id);
+    const agency = wishcard.belongsTo;
+    if (res.locals.user.userRole !== 'admin') {
+      const userAgency = await AgencyRepository.getAgencyByUserId(res.locals.user._id);
+      if (String(agency._id) !== String(userAgency._id)) {
+        return res.status(404).render('403');
+      }
+    }
+
+    await WishCardRepository.deleteWishCard(wishcard._id);
+
+    let url = '/wishcards/me';
+    if (res.locals.user.userRole === 'admin') {
+      url = '/wishcards/';
+    }
+
```

Comment:
you can move this before the first ```res.locals.user.userRole``` and just change the logic a bit so that you assign the admin route first, than check for userRole != admin and if that's true append the ```/me``` to the route

gets rid of the second userRole check